tests/conftest.py

```diff
@@ -537,3 +537,26 @@ def disable_enhanced_cancellation():
 @pytest.fixture
 def start_of_test() -> pendulum.DateTime:
     return pendulum.now("UTC")
+
+
+@pytest.fixture(autouse=True)
+def reset_sys_modules():
+    import importlib
+
+    original_modules = sys.modules.copy()
+
+    # Workaround for weird behavior on Linux where some of our "expected
+    # failure" tests succeed because '.' is in the path.
+    if sys.platform == "linux" and "." in sys.path:
+        sys.path.remove(".")
+
+    yield
+
+    # Delete all of the module objects that were introduced so they are not
+    # cached.
+    for module in set(sys.modules.keys()):
+        if module not in original_modules:
+            del sys.modules[module]
+
+    importlib.invalidate_caches()
+    sys.modules = original_modules
```

Comment:
Curious if this is necessary since we're already `del`ing the newly imported ones?

Reply:
Oh possibly not, let me test it.

Reply:
Nope, doesn't appear to be needed.